MapleServer2/PacketHandlers/Game/MeretMarketHandler.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            session.Send(MeretMarketPacket.Initialize());
+        }
+
+        private static void HandleOpenPremium(GameSession session, PacketReader packet)
+        {
+            MeretMarketCategory category = (MeretMarketCategory) packet.ReadInt();
+            List<MeretMarketMetadata> marketItems = MeretMarketMetadataStorage.GetCategoryItems(category);
+            if (marketItems == null)
+            {
+                return;
+            }
+            session.Send(MeretMarketPacket.Premium(marketItems));
+        }
+
+        private static void HandlePurchase(GameSession session, PacketReader packet)
+        {
+            byte quantity = packet.ReadByte();
+            int marketItemId = packet.ReadInt();
+            byte[] unk = packet.Read(12);
+            int childMarketItemId = packet.ReadInt();
+            long unk2 = packet.ReadLong();
+            int itemIndex = packet.ReadInt();
+            int totalQuantity = packet.ReadInt();
+            int unk3 = packet.ReadInt();
+            byte unk4 = packet.ReadByte();
+            string unk5 = packet.ReadUnicodeString();
+            string unk6 = packet.ReadUnicodeString();
+            long price = packet.ReadLong();
+
+            MeretMarketMetadata marketItem = MeretMarketMetadataStorage.GetMetadata(marketItemId);
+            if (marketItem == null)
+            {
+                return;
+            }
+
+            if (childMarketItemId == 0)
+            {
+
+                HandleMarketItemPay(session, marketItem, itemIndex, totalQuantity);
+            }
+            else
+            {
+                MeretMarketMetadata childItem = marketItem.AdditionalQuantities.FirstOrDefault(x => x.MarketItemId == childMarketItemId);
+                HandleMarketItemPay(session, childItem, itemIndex, totalQuantity);
+            }
+        }
+
+        private static void HandleMarketItemPay(GameSession session, MeretMarketMetadata marketItem, int itemIndex, int totalQuantity)
+        {
+            switch (marketItem.TokenType)
+            {
+                case MeretMarketCurrencyType.Meret:
+                    if (!session.Player.Wallet.RemoveMerets(marketItem.SalePrice))
+                    {
+                        return;
+                    }
+                    break;
+                case MeretMarketCurrencyType.Meso:
+                    if (!session.Player.Wallet.Meso.Modify(marketItem.SalePrice))
+                    {
+                        return;
+                    }
+                    break;
+            }
+
+            Item item = new Item(marketItem.ItemId)
+            {
+                Amount = marketItem.Quantity + marketItem.BonusQuantity,
+                Rarity = marketItem.Rarity
+            };
+            if (marketItem.Duration != 0)
+            {
+                item.ExpiryTime = DateTimeOffset.UtcNow.ToUnixTimeSeconds() + Environment.TickCount + (marketItem.Duration * 86400);
```

Comment:
Can the 86400 be broken down into further smaller multiplication so its more obvious what it represents?